include/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.hpp

```diff
@@ -0,0 +1,251 @@
+using ostk::mathematics::object::Vector3d;
+using ostk::mathematics::curvefitting::Interpolator;
+
+using ostk::physics::coordinate::Frame;
+using ostk::physics::data::Scalar;
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
+using ostk::physics::Unit;
+using ostk::physics::unit::Derived;
+using ostk::physics::unit::Length;
+using ostk::physics::unit::Mass;
+using ostk::physics::unit::Time;
+
+using TabulatedDynamics = ostk::astro::dynamics::Tabulated;
+using ostk::astro::flight::system::PropulsionSystem;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianVelocity;
+
+#define DEFAULT_MANEUVER_INTERPOLATION_TYPE Interpolator::Type::BarycentricRational
+
+/// @brief Stores information and metadata about a maneuver as performed by a spacecraft.
+class Maneuver
+{
+   public:
+    static const Shared<const Frame> DefaultAccelFrameSPtr;
+
+    /// @brief Constructor
+    ///
+    /// @param anInstantArray An array of instants, must be sorted
+    /// @param anAccelerationProfile An acceleration profile of the maneuver, one Vector3d per instant
+    /// @param aFrameSPtr A frame in which the acceleration profile is defined
+    /// @param aMassFlowRateProfile A mass flow rate profile of the maneuver, one Real per instant
+    Maneuver(
+        const Array<Instant>& anInstantArray,
+        const Array<Vector3d>& anAccelerationProfile,
+        const Shared<const Frame>& aFrameSPtr,
+        const Array<Real>& aMassFlowRateProfile
+    );
+
+    /// @brief Output stream operator
+    ///
+    /// @code{.cpp}
+    ///                  std::cout << Maneuver(...) ;
+    /// @endcode
+    ///
+    /// @param anOutputStream An output stream
+    /// @param aManeuver A Maneuver
+    ///
+    /// @return A reference to output stream
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const Maneuver& aManeuver);
+
+    /// @brief Check if the maneuver is defined
+    ///
+    /// @return True if maneuver is defined (always returns true)
+    bool isDefined() const;
+
+    /// @brief Get the instants of the maneuver
+    /// @return The instants
+    Array<Instant> getInstants() const;
+
+    /// @brief Get the acceleration profile of the maneuver
+    ///
+    /// @param aFrameSPtr A frame in which the acceleration profile is to be defined
+    ///
+    /// @return The acceleration profile
+    Array<Vector3d> getAccelerationProfile(const Shared<const Frame>& aFrameSPtr = DefaultAccelFrameSPtr) const;
+
+    /// @brief Get the mass flow rate profile of the maneuver
+    /// @return The mass flow rate profile
+    Array<Real> getMassFlowRateProfile() const;
+
+    /// @brief Get the start instant of the maneuver
+    /// @return The start instant
+    Instant getStartInstant() const;
+
+    /// @brief Get the end instant of the maneuver
+    /// @return The end instant
+    Instant getEndInstant() const;
```

Comment:
Let's return an interval instead of start and end to match patterns with some other objects.

Reply:
Sure, may be worth aligning `Segment` and `Sequence` with this paradigm as well then